extensions/amp-position-observer/0.1/amp-position-observer.js

```diff
@@ -450,6 +460,18 @@ export class AmpVisibilityObserver extends AMP.BaseElement {
       );
     }
   }
+
+  /**
+   * @private
+   */
+  maybeUninstallPositionObserver_() {
+    if (this.positionObserver_) {
```

Comment:
we can only do this if there is not `enter` and `exist` events handlers installed. So we sort of need a `if (scroll is the only event author has registered for), then ..`

ActionService has a `hasAction` that can be used to find what page author has attached to solve this. (Let's also move all the events this fire into a single const so we can iterate it here to avoid knowing about `enter` and `exist` in this logic)

Reply:
Done! Good shout I forgot about that case!